geometric/tests/test_hessian.py

```diff
@@ -42,6 +44,69 @@ def test_hessian_assort():
     assert np.allclose(fgrad_dlc, agrad_dlc, atol=1.e-6)
     assert np.allclose(fhess_dlc, ahess_dlc, atol=1.e-6)
 
+
+@addons.using_bigchem_psi4
+def test_psi4_bigchem_hessian():
+    # Turning on BigChem
+    geometric.nifty.CheckBigChem('psi4')
+    assert geometric.nifty.BigChemReady()
+
+    molecule, engine = geometric.prepare.get_molecule_engine(engine="psi4", input=os.path.join(datad, 'hcn_minimized.psi4in'))
+    coords = molecule.xyzs[0].flatten()*geometric.nifty.ang2bohr
+    hessian = geometric.normal_modes.calc_cartesian_hessian(coords, molecule, engine, tempfile.mkdtemp())
+    freqs, modes, G = geometric.normal_modes.frequency_analysis(coords, hessian, elem=molecule.elem, verbose=True)
+
+    np.testing.assert_almost_equal(freqs, [989.5974, 989.5992, 2394.0352, 3690.5745], decimal=0)
+    assert len(freqs) == 4
+
+    # Turning off BigChem
+    geometric.nifty.BigChemOff()
+    assert not geometric.nifty.BigChemReady()
```

Comment:
Strongly suggest you do not rely upon global switches! These lead to great unhappiness! Update function signature that use BigChem to have the `bigchem: bool = False` keyword argument instead to make BigChem explicit throughout your system.